### DISCUSSION TITLE
Mandatory ChangedSectionIdentifier and ChangedNoticeIdentifier nodes

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#675] [Q&A] [answered]
Is the Change SectionIdentifier node required when submitting an eForm of type change? And the node ChangedNoticeIdentifier?

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/64310812-957a-4b41-8f5b-23c7253da477)

We have validated using the validate operation of web service eForms of type change without the ChangedSectionIdentifier and ChangedNoticeIdentifier nodes and it is considered valid.

Thanks in advance

**Comment by rouschr** [comment, marked as answer]
You can have a look at the fields.json file, it does not look to be mandatory, but it is possible that we will have rules for this later.
```
{
    "id" : "BT-13716-notice",
    "parentNodeId" : "ND-ChangedSection",
    "name" : "Change Previous Section Identifier",
    "btId" : "BT-13716",
    "xpathAbsolute" : "/*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:Changes/efac:Change/efac:ChangedSection/efbc:ChangedSectionIdentifier",
    "xpathRelative" : "efbc:ChangedSectionIdentifier",
    "xsdSequenceOrder" : [ { "efbc:ChangedSectionIdentifier" : 1 } ],
    "type" : "id",
    "legalType" : "IDENTIFIER",
    "repeatable" : {
      "value" : false,
      "severity" : "ERROR"
    }, <--------------------------------------------- nothing about mandatory
    "forbidden" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "X01", "X02" ], <------------ forbidden in some notice sub types
        "value" : true,
        "severity" : "ERROR"
      } ]
    },
    "pattern" : {
      "value" : "^(PROCEDURE|BUYER|RESULT|((PAR|LOT|GLO|RES|ORG)-\\d{4}))$",
      "severity" : "ERROR"
    }
  }
```

**Reply by rouschr** [reply]
@YvesJo can you say something about the rules?

**Reply by YvesJo** [reply]
Hi,
To be able to fill in the change section, it is necessary to first identify the changed notice (BT-758). There is currently no rule in place that would mandate the use of BT-13716. Until decision is made to have such rules in place, it is up to the author to specify the impacted section, which I recommend. The relevance of this information increases with the size of the notice, there are very few cases where this wouldn't be of interest (cancellation of a notice, cancellation of a whole procedure). Also the procedure cancellation will have to be performed using a Result notice with non-awarded lots and the change notice with the "cancel-intent" code will most likely never be used.
